satpy/readers/ici_l1b_nc.py

```diff
@@ -0,0 +1,441 @@
+            longitude: xarray DataArray containing the longitude dataset to
+                interpolate.
+            latitude: xarray DataArray containing the longitude dataset to
+                interpolate.
+            n_samples: int describing number of samples per scan to interpolate
+                onto.
+
+        Returns:
+            tuple of arrays containing the interpolate values, all the original
+                metadata and the updated dimension names.
+
+        """
+        horns = longitude.n_horns
+        n_scan = longitude.n_scan
+        n_subs = longitude.n_subs
+        lons = np.zeros((n_scan.size, n_samples, horns.size))
+        lats = np.zeros((n_scan.size, n_samples, horns.size))
+        n_subs = np.linspace(0, n_samples - 1, n_subs.size).astype(int)
+        for horn in horns.values:
+            satint = GeoInterpolator(
+                (longitude.values[:, :, horn], latitude.values[:, :, horn]),
+                (n_scan.values, n_subs),
+                (n_scan.values, np.arange(n_samples)),
+            )
+            lons_horn, lats_horn = satint.interpolate()
+            lons[:, :, horn] = lons_horn
+            lats[:, :, horn] = lats_horn
+        lon = xr.DataArray(
+            lons,
+            attrs=longitude.attrs,
+            dims=['y', 'x', 'n_horns'],
+            coords={"n_horns": horns},
+        )
+        lat = xr.DataArray(
+            lats,
+            attrs=latitude.attrs,
+            dims=['y', 'x', 'n_horns'],
+            coords={"n_horns": horns},
+        )
+        return lon, lat
+
+    def _perform_viewing_angle_interpolation(
+        self,
+        azimuth,
+        zenith,
+        n_samples,
+    ):
+        """
+        Perform the interpolation of angular coordinates from tie points to pixel points.
+
+        Args:
+            azimuth: xarray DataArray containing the azimuth angle dataset to
+                interpolate.
+            zenith: xarray DataArray containing the zenith angle dataset to
+                interpolate.
+            n_samples: int describing number of samples per scan to interpolate
+                onto.
+
+        Returns:
+            tuple of arrays containing the interpolate values, all the original
+                metadata and the updated dimension names.
+
+        """
+        # transform to coordinates where origin is at the equator
+        x, y, z = lonlat2xyz(azimuth, 90. - zenith)
+        lon, lat = xyz2lonlat(x, y, z)
+        lon, lat = self._perform_geo_interpolation(
+            lon,
+            xr.DataArray(lat),
+            n_samples,
+        )
+        x, y, z = lonlat2xyz(lon, lat)
+        # transform from cartesian to spherical coords following Eumetsat spec
+        aa = np.degrees(np.arctan2(y, x))
+        za = np.degrees(np.arctan2(np.sqrt(x ** 2 + y ** 2), z))
+        aa.attrs = azimuth.attrs
+        za.attrs = zenith.attrs
+        return aa, za
+
+    def _perform_calibration(self, variable, dataset_info):
```

Comment:
Many functions here are called `perform_something`, could this be replace by a verb for clarity/brevity? (eg here: `_calibrate` would be enough.

Reply:
Thanks. I changed the function names according to your suggestion.